tests/jballerina-unit-test/src/test/resources/test-src/types/readonly/test_inherently_immutable_type.bal

```diff
@@ -163,6 +163,12 @@ function testRuntimeIsTypeForInherentlyImmutableBasicTypes() {
     'xml:Text xmlText = xml `xml text`;
     any n = xmlText;
     assertTrue(n is readonly);
+    
+    any reg1 = re `pattern`;
+    assertTrue(reg1 is readonly);
+    
+    readonly reg2 = re `pattern`;
+    assertTrue(reg2 is readonly);
```

Comment:
I would add a test with something like `regexp:fromString(str)` where the value is created at runtime with no compile-time input.